<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>CMSC33750 - Homework 1</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CMSC33750</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">CMSC33750 - Homework 1</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CMSC33750/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201001code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201001)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201001code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201001)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjovianyuCMSC33750tree8fc4415dbb6d9bd69268d37b037ce379fa828a1ftargetblank8fc4415a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jovianyu/CMSC33750/tree/8fc4415dbb6d9bd69268d37b037ce379fa828a1f" target="_blank">8fc4415</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjovianyuCMSC33750tree8fc4415dbb6d9bd69268d37b037ce379fa828a1ftargetblank8fc4415a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jovianyu/CMSC33750/tree/8fc4415dbb6d9bd69268d37b037ce379fa828a1f" target="_blank">8fc4415</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  GDCdata/

Unstaged changes:
    Modified:   analysis/hw1.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/hw1.Rmd</code>) and HTML (<code>docs/hw1.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jovianyu/CMSC33750/blob/8fc4415dbb6d9bd69268d37b037ce379fa828a1f/analysis/hw1.Rmd" target="_blank">8fc4415</a>
</td>
<td>
Jovian Yu
</td>
<td>
2020-10-01
</td>
<td>
Initial commit, first homework assignment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jovianyu/CMSC33750/8fc4415dbb6d9bd69268d37b037ce379fa828a1f/docs/hw1.html" target="_blank">8fc4415</a>
</td>
<td>
Jovian Yu
</td>
<td>
2020-10-01
</td>
<td>
Initial commit, first homework assignment
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<ul>
<li>Loading necessary libraries</li>
</ul>
</div>
<div id="downloading-gdc-data" class="section level1">
<h1>Downloading GDC data</h1>
<ul>
<li>TCGA head and neck cohort only</li>
</ul>
</div>
<div id="data-exploration" class="section level1">
<h1>Data Exploration</h1>
<ul>
<li>Listing all (clinical) metadata - some of this data is not particularly interesting/is missing, but some of the clinical data (extracted here) is meaningful</li>
<li>Using history of alcohol use as a primary variable to examine here ** A breakdown of stage vs alcohol use (removing data where it isn’t reported) ** Survival curves based on alcohol use - no clear differences here</li>
</ul>
<pre class="r"><code>print(colnames(colData(data)))</code></pre>
<pre><code> [1] &quot;barcode&quot;                     &quot;patient&quot;                    
 [3] &quot;sample&quot;                      &quot;shortLetterCode&quot;            
 [5] &quot;definition&quot;                  &quot;sample_submitter_id&quot;        
 [7] &quot;state&quot;                       &quot;submitter_id&quot;               
 [9] &quot;longest_dimension&quot;           &quot;intermediate_dimension&quot;     
[11] &quot;pathology_report_uuid&quot;       &quot;shortest_dimension&quot;         
[13] &quot;initial_weight&quot;              &quot;sample_id&quot;                  
[15] &quot;sample_type_id&quot;              &quot;sample_type&quot;                
[17] &quot;days_to_collection&quot;          &quot;tissue_type&quot;                
[19] &quot;oct_embedded&quot;                &quot;is_ffpe&quot;                    
[21] &quot;tumor_stage&quot;                 &quot;prior_malignancy&quot;           
[23] &quot;age_at_diagnosis&quot;            &quot;diagnosis_id&quot;               
[25] &quot;days_to_diagnosis&quot;           &quot;icd_10_code&quot;                
[27] &quot;synchronous_malignancy&quot;      &quot;tumor_grade&quot;                
[29] &quot;prior_treatment&quot;             &quot;primary_diagnosis&quot;          
[31] &quot;days_to_last_follow_up&quot;      &quot;site_of_resection_or_biopsy&quot;
[33] &quot;classification_of_tumor&quot;     &quot;ajcc_clinical_n&quot;            
[35] &quot;ajcc_clinical_stage&quot;         &quot;morphology&quot;                 
[37] &quot;ajcc_clinical_t&quot;             &quot;year_of_diagnosis&quot;          
[39] &quot;ajcc_pathologic_n&quot;           &quot;tissue_or_organ_of_origin&quot;  
[41] &quot;ajcc_staging_system_edition&quot; &quot;ajcc_pathologic_stage&quot;      
[43] &quot;ajcc_clinical_m&quot;             &quot;ajcc_pathologic_t&quot;          
[45] &quot;last_known_disease_status&quot;   &quot;progression_or_recurrence&quot;  
[47] &quot;treatments&quot;                  &quot;ajcc_pathologic_m&quot;          
[49] &quot;alcohol_history&quot;             &quot;cigarettes_per_day&quot;         
[51] &quot;exposure_id&quot;                 &quot;years_smoked&quot;               
[53] &quot;pack_years_smoked&quot;           &quot;days_to_death&quot;              
[55] &quot;vital_status&quot;                &quot;year_of_birth&quot;              
[57] &quot;race&quot;                        &quot;ethnicity&quot;                  
[59] &quot;age_at_index&quot;                &quot;year_of_death&quot;              
[61] &quot;gender&quot;                      &quot;days_to_birth&quot;              
[63] &quot;demographic_id&quot;              &quot;bcr_patient_barcode&quot;        
[65] &quot;name&quot;                        &quot;released&quot;                   
[67] &quot;project_id&quot;                  &quot;disease_type&quot;               
[69] &quot;releasable&quot;                  &quot;primary_site&quot;               
[71] &quot;paper_patient&quot;               &quot;paper_RNA&quot;                  
[73] &quot;paper_Methylation&quot;           &quot;paper_RPPA&quot;                 
[75] &quot;paper_miRNA&quot;                 &quot;paper_Copy.Number&quot;          
[77] &quot;paper_PARADIGM&quot;             </code></pre>
<pre class="r"><code>tib_alc_vs_stage &lt;- tibble(Alcohol = factor(data$alcohol_history), Stage = factor(data$tumor_stage)) %&gt;%
  filter(Alcohol != &quot;Not Reported&quot;, Stage != &quot;not reported&quot;) %&gt;%
  group_by(Alcohol, Stage) %&gt;%
  summarise(n = n())</code></pre>
<pre><code>`summarise()` regrouping output by &#39;Alcohol&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code>tib_alc_vs_stage %&gt;%
  ggplot(aes(x = Alcohol, y = Stage, fill = n)) +
  geom_raster()</code></pre>
<p><img src="figure/hw1.Rmd/Exploration-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tib_survival_alcohol &lt;- tibble(
    days_to_death = data$days_to_death,
    days_to_last_follow_up = data$days_to_last_follow_up,
    vital_status = factor(data$vital_status),
    alcohol = factor(data$alcohol_history)
  ) %&gt;%
  mutate(
    vital_status = ifelse(vital_status == &quot;Alive&quot;, 0, 1),
    years = as.numeric(ifelse(vital_status == 1, days_to_death, days_to_last_follow_up))/365.0,
    alcohol = ifelse(alcohol == &quot;Yes&quot;, 1, ifelse(alcohol == &quot;No&quot;, 0, NA_character_))
  ) %&gt;%
  select(vital_status, alcohol, years) %&gt;%
  na.omit()

fit_survival_alcohol &lt;- survfit(Surv(years, vital_status) ~ alcohol, data = tib_survival_alcohol)

ggsurvplot(fit_survival_alcohol, data = tib_survival_alcohol, risk.table = TRUE,
           legend.labs = c(&quot;No Alcohol History&quot;, &quot;Alcohol History&quot;),
           xlab = &quot;Survival (Years)&quot;)</code></pre>
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
Results may be unexpected or may change in future versions of ggplot2.</code></pre>
<p><img src="figure/hw1.Rmd/Exploration-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="transcript-counts" class="section level1">
<h1>Transcript Counts</h1>
<ul>
<li>Showing counts here, subsetting on the first on only patients with Stage IVB disease ** Generates a reasonably sized subset of 11 patients with an interesting phenotype</li>
<li>Plotting with ggbio (to allow for direct manipulation of a SummarizedExperiment object with ggplot) with a log transform of both axes to make the visualization reasonable</li>
</ul>
<pre class="r"><code>list_stageivb &lt;- data$tumor_stage == &quot;stage ivb&quot;

print(summary(list_stageivb))</code></pre>
<pre><code>   Mode   FALSE    TRUE 
logical     535      11 </code></pre>
<pre class="r"><code>data[, list_stageivb] %&gt;%
  ggplot(aes(x = value), assay.id = 1L, environment = parent.frame()) +
  geom_histogram(na.rm = TRUE) +
  scale_x_log10(
   breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x),
   labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))
  ) +
  scale_y_log10(
   breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x),
   labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))
  ) +
  annotation_logticks() +
  theme_bw() +
  labs(title = &quot;Histogram of RNA Transcript Counts in the\nTCGA H&amp;N Patients with Stage IVB Disease&quot;) + xlab(&quot;Number of Transcripts&quot;) + ylab(&quot;Number of Genes&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/hw1.Rmd/Histogram%20of%20Counts-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<pre class="r"><code>sessioninfo::package_info()</code></pre>
<pre><code> package              * version  date       lib source        
 abind                  1.4-5    2016-07-21 [1] CRAN (R 4.0.0)
 AnnotationDbi          1.50.3   2020-07-25 [1] Bioconductor  
 AnnotationFilter       1.12.0   2020-04-27 [1] Bioconductor  
 askpass                1.1      2019-01-13 [1] CRAN (R 4.0.0)
 assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.0.0)
 backports              1.1.10   2020-09-15 [1] CRAN (R 4.0.2)
 base64enc              0.1-3    2015-07-28 [1] CRAN (R 4.0.0)
 Biobase              * 2.48.0   2020-04-27 [1] Bioconductor  
 BiocFileCache          1.12.1   2020-08-04 [1] Bioconductor  
 BiocGenerics         * 0.34.0   2020-04-27 [1] Bioconductor  
 BiocManager          * 1.30.10  2019-11-16 [1] CRAN (R 4.0.2)
 BiocParallel           1.22.0   2020-04-27 [1] Bioconductor  
 biomaRt                2.44.1   2020-06-17 [1] Bioconductor  
 Biostrings             2.56.0   2020-04-27 [1] Bioconductor  
 biovizBase             1.36.0   2020-04-27 [1] Bioconductor  
 bit                    4.0.4    2020-08-04 [1] CRAN (R 4.0.2)
 bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.0.2)
 bitops                 1.0-6    2013-08-17 [1] CRAN (R 4.0.0)
 blob                   1.2.1    2020-01-20 [1] CRAN (R 4.0.0)
 broom                  0.7.0    2020-07-09 [1] CRAN (R 4.0.2)
 BSgenome               1.56.0   2020-04-27 [1] Bioconductor  
 car                    3.0-10   2020-09-29 [1] CRAN (R 4.0.2)
 carData                3.0-4    2020-05-22 [1] CRAN (R 4.0.0)
 cellranger             1.1.0    2016-07-27 [1] CRAN (R 4.0.0)
 checkmate              2.0.0    2020-02-06 [1] CRAN (R 4.0.2)
 cli                    2.0.2    2020-02-28 [1] CRAN (R 4.0.0)
 cluster                2.1.0    2019-06-19 [2] CRAN (R 4.0.2)
 colorspace             1.4-1    2019-03-18 [1] CRAN (R 4.0.0)
 crayon                 1.3.4    2017-09-16 [1] CRAN (R 4.0.0)
 curl                   4.3      2019-12-02 [1] CRAN (R 4.0.0)
 data.table             1.13.0   2020-07-24 [1] CRAN (R 4.0.2)
 DBI                    1.1.0    2019-12-15 [1] CRAN (R 4.0.0)
 dbplyr                 1.4.4    2020-05-27 [1] CRAN (R 4.0.0)
 DelayedArray         * 0.14.1   2020-07-15 [1] Bioconductor  
 dichromat              2.0-0    2013-01-24 [1] CRAN (R 4.0.0)
 digest                 0.6.25   2020-02-23 [1] CRAN (R 4.0.0)
 downloader             0.4      2015-07-09 [1] CRAN (R 4.0.2)
 dplyr                * 1.0.2    2020-08-18 [1] CRAN (R 4.0.2)
 DT                   * 0.15     2020-08-05 [1] CRAN (R 4.0.2)
 ellipsis               0.3.1    2020-05-15 [1] CRAN (R 4.0.0)
 ensembldb              2.12.1   2020-05-06 [1] Bioconductor  
 evaluate               0.14     2019-05-28 [1] CRAN (R 4.0.0)
 fansi                  0.4.1    2020-01-08 [1] CRAN (R 4.0.0)
 farver                 2.0.3    2020-01-16 [1] CRAN (R 4.0.0)
 forcats              * 0.5.0    2020-03-01 [1] CRAN (R 4.0.0)
 foreign                0.8-80   2020-05-24 [2] CRAN (R 4.0.2)
 Formula                1.2-3    2018-05-03 [1] CRAN (R 4.0.0)
 fs                     1.5.0    2020-07-31 [1] CRAN (R 4.0.2)
 generics               0.0.2    2018-11-29 [1] CRAN (R 4.0.0)
 GenomeInfoDb         * 1.24.2   2020-06-15 [1] Bioconductor  
 GenomeInfoDbData       1.2.3    2020-04-28 [1] Bioconductor  
 GenomicAlignments      1.24.0   2020-04-27 [1] Bioconductor  
 GenomicFeatures        1.40.1   2020-07-08 [1] Bioconductor  
 GenomicRanges        * 1.40.0   2020-04-27 [1] Bioconductor  
 GGally                 2.0.0    2020-06-06 [1] CRAN (R 4.0.2)
 ggbio                * 1.36.0   2020-04-27 [1] Bioconductor  
 ggplot2              * 3.3.2    2020-06-19 [1] CRAN (R 4.0.2)
 ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.0.2)
 ggsignif               0.6.0    2019-08-08 [1] CRAN (R 4.0.2)
 git2r                  0.27.1   2020-05-03 [1] CRAN (R 4.0.0)
 glue                   1.4.2    2020-08-27 [1] CRAN (R 4.0.2)
 graph                  1.66.0   2020-04-27 [1] Bioconductor  
 gridExtra              2.3      2017-09-09 [1] CRAN (R 4.0.0)
 gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.0.0)
 haven                  2.3.1    2020-06-01 [1] CRAN (R 4.0.0)
 Hmisc                  4.4-1    2020-08-10 [1] CRAN (R 4.0.0)
 hms                    0.5.3    2020-01-08 [1] CRAN (R 4.0.0)
 htmlTable              2.1.0    2020-09-16 [1] CRAN (R 4.0.2)
 htmltools              0.5.0    2020-06-16 [1] CRAN (R 4.0.2)
 htmlwidgets            1.5.1    2019-10-08 [1] CRAN (R 4.0.2)
 httpuv                 1.5.4    2020-06-06 [1] CRAN (R 4.0.0)
 httr                   1.4.2    2020-07-20 [1] CRAN (R 4.0.2)
 IRanges              * 2.22.2   2020-05-21 [1] Bioconductor  
 jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 4.0.0)
 jsonlite               1.7.1    2020-09-07 [1] CRAN (R 4.0.0)
 km.ci                  0.5-2    2009-08-30 [1] CRAN (R 4.0.2)
 KMsurv                 0.1-5    2012-12-03 [1] CRAN (R 4.0.0)
 knitr                  1.30     2020-09-22 [1] CRAN (R 4.0.2)
 labeling               0.3      2014-08-23 [1] CRAN (R 4.0.0)
 later                  1.1.0.1  2020-06-05 [1] CRAN (R 4.0.0)
 lattice                0.20-41  2020-04-02 [2] CRAN (R 4.0.2)
 latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.0.2)
 lazyeval               0.2.2    2019-03-15 [1] CRAN (R 4.0.0)
 lifecycle              0.2.0    2020-03-06 [1] CRAN (R 4.0.0)
 lubridate              1.7.9    2020-06-08 [1] CRAN (R 4.0.0)
 magrittr               1.5      2014-11-22 [1] CRAN (R 4.0.0)
 Matrix                 1.2-18   2019-11-27 [2] CRAN (R 4.0.2)
 matrixStats          * 0.57.0   2020-09-25 [1] CRAN (R 4.0.2)
 memoise                1.1.0    2017-04-21 [1] CRAN (R 4.0.0)
 modelr                 0.1.8    2020-05-19 [1] CRAN (R 4.0.0)
 munsell                0.5.0    2018-06-12 [1] CRAN (R 4.0.0)
 nnet                   7.3-14   2020-04-26 [2] CRAN (R 4.0.2)
 openssl                1.4.3    2020-09-18 [1] CRAN (R 4.0.2)
 openxlsx               4.2.2    2020-09-17 [1] CRAN (R 4.0.2)
 OrganismDbi            1.30.0   2020-04-28 [1] Bioconductor  
 pillar                 1.4.6    2020-07-10 [1] CRAN (R 4.0.2)
 pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.0.0)
 plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.0.0)
 png                    0.1-7    2013-12-03 [1] CRAN (R 4.0.0)
 prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.0.0)
 progress               1.2.2    2019-05-16 [1] CRAN (R 4.0.0)
 promises               1.1.1    2020-06-09 [1] CRAN (R 4.0.0)
 ProtGenerics           1.20.0   2020-04-27 [1] Bioconductor  
 purrr                * 0.3.4    2020-04-17 [1] CRAN (R 4.0.0)
 R.methodsS3            1.8.1    2020-08-26 [1] CRAN (R 4.0.2)
 R.oo                   1.24.0   2020-08-26 [1] CRAN (R 4.0.2)
 R.utils                2.10.1   2020-08-26 [1] CRAN (R 4.0.2)
 R6                     2.4.1    2019-11-12 [1] CRAN (R 4.0.0)
 rappdirs               0.3.1    2016-03-28 [1] CRAN (R 4.0.0)
 RBGL                   1.64.0   2020-04-27 [1] Bioconductor  
 RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.0.0)
 Rcpp                   1.0.5    2020-07-06 [1] CRAN (R 4.0.2)
 RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 4.0.0)
 readr                * 1.3.1    2018-12-21 [1] CRAN (R 4.0.0)
 readxl                 1.3.1    2019-03-13 [1] CRAN (R 4.0.0)
 reprex                 0.3.0    2019-05-16 [1] CRAN (R 4.0.0)
 reshape                0.8.8    2018-10-23 [1] CRAN (R 4.0.2)
 reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.0.0)
 rio                    0.5.16   2018-11-26 [1] CRAN (R 4.0.0)
 rlang                  0.4.7    2020-07-09 [1] CRAN (R 4.0.2)
 rmarkdown              2.4      2020-09-30 [1] CRAN (R 4.0.2)
 rpart                  4.1-15   2019-04-12 [2] CRAN (R 4.0.2)
 rprojroot              1.3-2    2018-01-03 [1] CRAN (R 4.0.0)
 Rsamtools              2.4.0    2020-04-27 [1] Bioconductor  
 RSQLite                2.2.1    2020-09-30 [1] CRAN (R 4.0.2)
 rstatix                0.6.0    2020-06-18 [1] CRAN (R 4.0.2)
 rstudioapi             0.11     2020-02-07 [1] CRAN (R 4.0.0)
 rtracklayer            1.48.0   2020-04-27 [1] Bioconductor  
 rvest                  0.3.6    2020-07-25 [1] CRAN (R 4.0.2)
 S4Vectors            * 0.26.1   2020-05-16 [1] Bioconductor  
 scales                 1.1.1    2020-05-11 [1] CRAN (R 4.0.0)
 sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 4.0.0)
 stringi                1.5.3    2020-09-09 [1] CRAN (R 4.0.2)
 stringr              * 1.4.0    2019-02-10 [1] CRAN (R 4.0.0)
 SummarizedExperiment * 1.18.2   2020-07-09 [1] Bioconductor  
 survival             * 3.2-7    2020-09-28 [1] CRAN (R 4.0.2)
 survminer            * 0.4.8    2020-07-25 [1] CRAN (R 4.0.2)
 survMisc               0.5.5    2018-07-05 [1] CRAN (R 4.0.2)
 TCGAbiolinks         * 2.16.4   2020-09-08 [1] Bioconductor  
 tibble               * 3.0.3    2020-07-10 [1] CRAN (R 4.0.2)
 tidyr                * 1.1.2    2020-08-27 [1] CRAN (R 4.0.2)
 tidyselect             1.1.0    2020-05-11 [1] CRAN (R 4.0.0)
 tidyverse            * 1.3.0    2019-11-21 [1] CRAN (R 4.0.2)
 VariantAnnotation      1.34.0   2020-04-27 [1] Bioconductor  
 vctrs                  0.3.4    2020-08-29 [1] CRAN (R 4.0.2)
 whisker                0.4      2019-08-28 [1] CRAN (R 4.0.0)
 withr                  2.3.0    2020-09-22 [1] CRAN (R 4.0.2)
 workflowr            * 1.6.2    2020-04-30 [1] CRAN (R 4.0.2)
 xfun                   0.17     2020-09-09 [1] CRAN (R 4.0.2)
 XML                    3.99-0.5 2020-07-23 [1] CRAN (R 4.0.2)
 xml2                   1.3.2    2020-04-23 [1] CRAN (R 4.0.0)
 xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.0.0)
 XVector                0.28.0   2020-04-27 [1] Bioconductor  
 yaml                   2.2.1    2020-02-01 [1] CRAN (R 4.0.0)
 zip                    2.1.1    2020-08-27 [1] CRAN (R 4.0.2)
 zlibbioc               1.34.0   2020-04-27 [1] Bioconductor  
 zoo                    1.8-8    2020-05-02 [1] CRAN (R 4.0.0)

[1] Z:/Documents/R/win-library/4.0
[2] C:/Program Files/R/R-4.0.2/library</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    
system code page: 932

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggbio_1.36.0                survminer_0.4.8            
 [3] ggpubr_0.4.0                survival_3.2-7             
 [5] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
 [7] matrixStats_0.57.0          Biobase_2.48.0             
 [9] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
[11] IRanges_2.22.2              S4Vectors_0.26.1           
[13] BiocGenerics_0.34.0         DT_0.15                    
[15] forcats_0.5.0               stringr_1.4.0              
[17] dplyr_1.0.2                 purrr_0.3.4                
[19] readr_1.3.1                 tidyr_1.1.2                
[21] tibble_3.0.3                ggplot2_3.3.2              
[23] tidyverse_1.3.0             TCGAbiolinks_2.16.4        
[25] BiocManager_1.30.10         workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             backports_1.1.10         Hmisc_4.4-1             
  [4] BiocFileCache_1.12.1     plyr_1.8.6               lazyeval_0.2.2          
  [7] splines_4.0.2            BiocParallel_1.22.0      digest_0.6.25           
 [10] ensembldb_2.12.1         htmltools_0.5.0          fansi_0.4.1             
 [13] checkmate_2.0.0          magrittr_1.5             memoise_1.1.0           
 [16] BSgenome_1.56.0          cluster_2.1.0            openxlsx_4.2.2          
 [19] Biostrings_2.56.0        modelr_0.1.8             R.utils_2.10.1          
 [22] askpass_1.1              prettyunits_1.1.1        jpeg_0.1-8.1            
 [25] colorspace_1.4-1         blob_1.2.1               rvest_0.3.6             
 [28] rappdirs_0.3.1           haven_2.3.1              xfun_0.17               
 [31] crayon_1.3.4             RCurl_1.98-1.2           jsonlite_1.7.1          
 [34] graph_1.66.0             VariantAnnotation_1.34.0 zoo_1.8-8               
 [37] glue_1.4.2               gtable_0.3.0             zlibbioc_1.34.0         
 [40] XVector_0.28.0           car_3.0-10               abind_1.4-5             
 [43] scales_1.1.1             DBI_1.1.0                GGally_2.0.0            
 [46] rstatix_0.6.0            Rcpp_1.0.5               xtable_1.8-4            
 [49] progress_1.2.2           htmlTable_2.1.0          foreign_0.8-80          
 [52] bit_4.0.4                OrganismDbi_1.30.0       km.ci_0.5-2             
 [55] Formula_1.2-3            htmlwidgets_1.5.1        httr_1.4.2              
 [58] RColorBrewer_1.1-2       ellipsis_0.3.1           farver_2.0.3            
 [61] pkgconfig_2.0.3          reshape_0.8.8            XML_3.99-0.5            
 [64] R.methodsS3_1.8.1        nnet_7.3-14              dbplyr_1.4.4            
 [67] labeling_0.3             tidyselect_1.1.0         rlang_0.4.7             
 [70] reshape2_1.4.4           later_1.1.0.1            AnnotationDbi_1.50.3    
 [73] munsell_0.5.0            cellranger_1.1.0         tools_4.0.2             
 [76] downloader_0.4           cli_2.0.2                generics_0.0.2          
 [79] RSQLite_2.2.1            broom_0.7.0              evaluate_0.14           
 [82] yaml_2.2.1               knitr_1.30               bit64_4.0.5             
 [85] fs_1.5.0                 zip_2.1.1                survMisc_0.5.5          
 [88] AnnotationFilter_1.12.0  RBGL_1.64.0              whisker_0.4             
 [91] R.oo_1.24.0              xml2_1.3.2               biomaRt_2.44.1          
 [94] compiler_4.0.2           rstudioapi_0.11          curl_4.3                
 [97] png_0.1-7                ggsignif_0.6.0           reprex_0.3.0            
[100] stringi_1.5.3            GenomicFeatures_1.40.1   lattice_0.20-41         
[103] ProtGenerics_1.20.0      Matrix_1.2-18            KMsurv_0.1-5            
[106] vctrs_0.3.4              pillar_1.4.6             lifecycle_0.2.0         
[109] data.table_1.13.0        bitops_1.0-6             rtracklayer_1.48.0      
[112] httpuv_1.5.4             R6_2.4.1                 latticeExtra_0.6-29     
[115] promises_1.1.1           gridExtra_2.3            rio_0.5.16              
[118] sessioninfo_1.1.1        dichromat_2.0-0          assertthat_0.2.1        
[121] openssl_1.4.3            rprojroot_1.3-2          withr_2.3.0             
[124] GenomicAlignments_1.24.0 Rsamtools_2.4.0          GenomeInfoDbData_1.2.3  
[127] hms_0.5.3                grid_4.0.2               rpart_4.1-15            
[130] rmarkdown_2.4            carData_3.0-4            biovizBase_1.36.0       
[133] git2r_0.27.1             lubridate_1.7.9          base64enc_0.1-3         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
